pkg/common/cns-lib/volume/manager.go

```diff
@@ -528,9 +574,19 @@ func (m *defaultManager) createVolumeWithImprovedIdempotency(ctx context.Context
 		// created on CNS.
 		if volumeOperationDetails != nil && volumeOperationDetails.OperationDetails != nil &&
 			volumeOperationDetails.OperationDetails.TaskStatus != taskInvocationStatusInProgress {
-			err := m.operationStore.StoreRequestDetails(ctx, volumeOperationDetails)
-			if err != nil {
-				log.Warnf("failed to store CreateVolume details with error: %v", err)
+
+			// Decrease the reserved field in QuotaDetails when the CreateVolume task is successful.
+			if volumeOperationDetails.OperationDetails.TaskStatus == taskInvocationStatusSuccess &&
+				volumeOperationDetails.QuotaDetails != nil {
+				volumeOperationDetails.QuotaDetails.Reserved = resource.NewQuantity(0,
+					resource.BinarySI)
+				log.Infof("Setting the reserved field for VolumeOperationDetails instance %s to 0",
+					volumeOperationDetails.Name)
+			}
+			tempErr := m.operationStore.StoreRequestDetails(ctx, volumeOperationDetails)
+			if err == nil && tempErr != nil {
+				log.Errorf("failed to store CreateVolume details with error: %v", tempErr)
+				err = tempErr
```

Comment:
What will happen if StoreRequestDetails fails?  Will there be a retry?

Reply:
Yes, it will be retried. Same as we do in the multi VC setup.